\chapter{OVERVIEW}

\section{Introduction}

In the 2000s, the advent of new nucleotide sequencing strategies based on ion semiconductors (Ion Torrent), pyrosequencing (454), and sequencing-by-synthesis (Illumina) provided new tools for studying genomes of both model and non-model organisms at unprecedented scale, resolution, and cost effectiveness.
More recent platforms based on single-molecule sequencing (PacBio SMRT and Oxford Nanopore) promise to ... (continue to?)
By the 2010s, these so-called \textit{next-generation sequencing (NGS)} technologies\footnote{Asdf} had transformed genome sequencing into a routine laboratory procedure, leading to a tremendous increase in the number of published genome projects and draft genome sequences (cite NCBI genomes page).
To say that these new technologies have transformed the life sciences is an understatement.

The availability of high-quality model reference genomes in this same time frame, however, has increased at a much more modest pace.
The effort and resources required to refine a provisional draft genome into a complete, finished product remains substantial.
Data collection is now extremely cheap, and an abundance of new algorithms and software tools have emerged for analyzing NGS data.
However, the tasks of genome assembly and annotation remain complex and difficult (cite Assemblathons, others), due in no small part to NGS data's own inherent limitations.
As a result, data quality can vary considerably across data sets and even within the same data set.
Doing good, reproducible science in such a setting requires not only an awareness of these issues but also a framework for disciplined quality control and analysis of annotated genomes.
The focus of this dissertation has been the development of that framework (and associated software tools) as motivated by research problems I encountered in genomics research projects.

My first task as a graduate student was to evaluate the performance of a new automated genome annotation workflow, involving primarily the comparison of its outputs to an existing annotation.
Similar comparisons were also required in subsequent projects, particularly while re-annotating an insect genome with an improved transcriptome assembly and additional protein evidence.
Quantitative measures of agreement between gene predictions and a reliable reference had long been established (cite Burset/Guigo), and are easily adapted for comparison of two alternative sources of annotation of unknown relative quality.
However, software capable of computing these similarity statistics at the time (cite Eval, GFPE) were unsatisfactory for our needs.
Designed more for algorithm and model refinement than for biological interpretation, these tools report a huge number of statistics aggregated over all data inputs, offering exquisite detail into overall performance but no detail at the level of individual loci.
To address these limitations, I created the ParsEval tool to provide locus-level reports in addition to overall aggregate statistics.
The strategy utilized by ParsEval for partitioning the genome into units that can be independently analyzed not only offered significant improvements in runtime and memory usage, but also provided the foundation for developing a more generalized genome analysis framework.

Another one of my earliest tasks as a graduate student was to assist with \textit{de novo} assembly of the genome of the paper wasp \textit{Polistes dominula}.
I subsequently ended up taking charge of the assembly effort, as well as genome annotation, transcriptome profiling, and comparative genomics analysis.
Pursuit of these latter

\section{Dissertation Organization}

This dissertation is organized into six chapters.
Chapter 1 provides an overview of the dissertation, a motivation for the work, and a brief discussion of relevant literature.
Chapters 2 through 5 are presented as complete manuscripts.
Chapter 2 is a research paper published in \textit{Molecular Ecology} describing the genome, transcriptome, and methylome of the paper wasp \textit{Polistes dominula}, highlighting its reduced DNA methylation system, several hundred genomic loci with caste-related differential expression, and the lack of any detectable caste-related alternative splicing in the adult organism.
Chapter 3 is a paper published in \textit{BMC Bioinformatics} describing \textit{ParsEval}, a tool for comparing two alternative sources of annotation for a genome sequence.
Chapter 4 is a methodology paper currently under review at \textit{BMC Bioinformatics/Genomics}, describing the use of \textit{interval loci (iLoci)} as an organizational framework facilitating reproducible genome analysis.
Chapter 5 is a manuscript slated for submission to \textit{A Journal} describing \textit{GeneAnnoLogy}, an \textit{iLocus}-based tool for quality control and version control of genome annotations.
Chapter 6 provides brief concluding remarks and suggestions for further research.
