\chapter{CONCLUSIONS}

The focus of this dissertation has been the development of tools and methodologies for reproducible analysis of genome annotations, including and especially those associated with NGS-based genome projects.
The motivation for much of this work was provided by challenges encountered during the assembly, annotation, and analysis of the genome of the paper wasp \textit{Polistes dominula}.
In addressing these challenges, we sought to create tools that could be applied more generally to address common questions about genome composition, and organization.

In the \textit{P. dominula} genome project, we developed a variety of genomic data resources \textit{de novo} for investigating genome composition, gene expression, and characteristics of DNA methylation.
Our study revealed that the paper wasp genome is similar to other social insects across a variety of measures: genome size, number of annotated genes, and exon and intron characteristics, but that its nucleotide composition is the most AT-rich (aggregated over large genomic sequences) of any hymenopteran studied.
We identified 367 loci in the \textit{P. dominula} genome that are differentially expressed between queen wasps and worker wasps, and though we catalogued thousands of alternative splicing events we found no evidence of differential splicing between adult queens and workers.
Phylogenetic analysis of putative single-copy orthologs conserved in the three primary aculeate lineages (ants, bees, and vespid wasps) was unable to settle the unresolved evolutionary lineage of this clade.
Finally, we discovered that \textit{Polistes} lacks the \textit{Dnmt3} DNA methyltransferase enzyme and has essentially zero DNA methylation genome-wide, calling into question the conventional wisdom that DNA methylation is critical to the evolution of social behavior.
This finding was independently confirmed in a study of another wasp from the same genus, \textit{Polistes canadensis}, published just weeks before our study was submitted for peer review.

The ParsEval tool addresses the fundamental need for data comparison: a need we encountered while annotating the paper wasp genome, and a need which is common to all genome projects.
The similarity statistics reported by ParsEval help pinpoint the ways in which different sets of annotations differ, facilitating the annotation process and particularly the selection parameters.
The locus-level detail and graphical reports produced by ParsEval are particularly beneficial in interpreting differences in annotation sets.

Investigating genome composition, comparative genomics questions, gene expression, and DNA methylation in the \textit{P. dominula} genome project required precise and careful handling of genome annotation data.
The iLocus framework was developed initially to address these issues.
iLoci provide a straightforward mechanism for determining the proportion of the genome occupied by various types of elements, facilitating the comparison of gene content and overall genome composition across species.
iLoci also defined the coordinate system for our analysis of gene expression, specifying an unambiguous handling of overlapping gene models and unit of quality control.
We later generalized these concepts and applied them to a wide variety of model organisms, describing the range of genome composition and genome organization exhibited across eukaryotic diversity.
iLoci furnish a well-defined notion of genome ``compactness" that is particularly consistent between different genome sequences from the same species and between different species within the same clade.
Finally, iLoci provide stability between different assembly and annotation versions, facilitating reproducibility as genome assemblies and annotations are refined over time.

One direct extension of the work described in this dissertation is motivated by the reproducibility challenges introduced as annotated genome assemblies are improved over time.
Refinement of our \textit{Polistes dominula} assembly and annotation hindered progress on our research at times, and this issue is certainly not unique to our work.
For example, the honeybee \textit{Apis mellifera} (the model social insect) had three official annotation versions in concurrent use during our work on the \textit{Polistes} genome project, with some studies even making their own unpublished refinements to an annotation \cite{Dnmt3KD,TrueSight}.
What seemed to be lacking was a precise way to refer to a particular gene or genomic region, as annotated at a particular time, and to make statements about its expression, or conservation, or methylation status, or any number of additional characteristics.
The idea of tracking annotations over time is not new \cite{Eilbeck}, and some well-supported communities provide tools for mapping annotations from an older assembly version to updated assembly \cite{liftOver,PAGIT}.
iLoci provide an alternative solution to these issues and, along with the additional benefits previously described, furnish a complete framework for organization, quality control, and reproducibility for provisional genome projects.
I have prototyped a software tool for maintaining an annotation version history, leveraging existing version control tools to track changes to individual iLoci over time.
Supplementing this tool with quality control features for filtering iLoci based on their length, nucleotide composition, annotation quality, or other characteristics would provide a useful tool for selecting finely tuned data sets for analysis of genome features.
