\chapter{INTRODUCTION}

\section{Overview}

In the 2000s, the advent of new nucleotide sequencing strategies based on ion semiconductors (Ion Torrent), pyrosequencing (454), and sequencing-by-synthesis (Illumina) provided new tools for studying genomes of both model and non-model organisms at unprecedented scale, resolution, and cost effectiveness.
These technologies continue to evolve, the latest innovations involving single-molecule long read sequencing (Pacific Biosciences SMRT and Oxford Nanopore).
By the 2010s, these so-called \textit{next-generation sequencing (NGS)} technologies had made genome sequencing accessible to essentially any scientist with even a modest research budget.
This ``democratization of sequencing" has precipitated a tremendous increase in the number of published genome projects and draft genome sequences (\url{http://www.ncbi.nlm.nih.gov/genome/browse/}), as well as genome-scale data sets profiling gene expression, chromatin accessibility, transcription initiation, and a multitude of other genomic characteristics.

During this same time frame, however, the democratization of genome analysis has occurred to a much lesser extent.
The availability of high-quality model reference genomes has changed very little.
And as newly acquired data continues to flood into public databases, many scientists struggle to effectively manage the data and critically evaluate downstream research products.
The proliferation of new algorithms and software tools for analyzing NGS data is a mixed blessing for scientists who now have both the flexibility and the burden of selecting suitable tool(s) for a particular analysis.

The complexity and difficulty of genome assembly \cite{Assemblathon1,Assemblathon2,GAGE,QUAST}, annotation \cite{EGASP,AED,DentonFly}, and analysis \cite{PigWatson,RnaSeqTrim,RnaSeqTrim2} has been reported in various recent studies and community projects.
The consistent, resounding message from this growing body of work is that genomics data quality varies considerably across data sets (and even within a single data set), and that the performance of state-of-the-art algorithms is difficult to predict on new data.
Without a well-funded and well-staffed research consortium to manage the painstaking work of gap-filling each new genome assembly and carefully curating its contents, the new reality is that most reference genome assemblies will remain fragmented and unfinished, and that provisionally annotated genome features will fall on a wide spectrum of reliability.
Doing principled and reproducible science in this setting requires disciplined quality control and data evaluation.
The focus of this dissertation has been the development of a framework---and associated software tools---to enable robust annotation and analysis of NGS-based genomics data, as motivated by research problems I encountered in genomics research projects.

\section{\textit{Polistes dominula} genome project}

My first encounter with many of these issues came from my genomics studies of the paper wasp \textit{Polistes dominula}.
This wasp is an important model system for studying the evolution of social behavior, as it exhibits an intermediate level of social complexity, with no morphological differentiation between castes and frequent competition for reproductive opportunities within colonies \cite{JandtToth}.
The molecular mechanisms underlying caste differentiation across all social insects are still poorly understood, although various mechanisms have been proposed to play primary roles, such as differential gene expression \cite{ChenDiffExp,GrozingerDiffExp,WhitfieldDiffExp,OmettoDiffExp,SimolaDiffExp,HarrisonDiffExp,PcanTrans}, differential splicing \cite{Dnmt3KD}, DNA methylation patterns \cite{Dnmt3KD,Lyko2011}, and recently emerged species- or clade-specific genes \cite{SumnerTRG,JohnsonTRG}.
These hypotheses have been tested in bees and ants, but prior to this study no genome resources were available for any species from the major vespid wasp lineage.
This project was funded to sequence the genome, transcriptome, and methylome of the wasp, to facilitate investigation of the molecular basis of caste differentiation in \textit{Polistes}, and to provide an additional important data source for comparative analysis of all social insects.

Our initial work was driven largely by questions of genome composition.
How large is the \textit{Polistes} genome?
What is its nucleotide composition?
How many genes does it encode?
Are the small handful of well-known ``social behavior" genes present in the genome as expected?
In short, is there anything that immediately distinguishes the paper wasp genome from genomes of related species?

The next set of questions were driven by a comparative genomics perspective.
How does the size and composition of the \textit{Polistes} genome compare to the bees and the ants?
What proportion of annotated gene models are well-conserved within the Hymenoptera?
Can conserved single-copy orthologs provide any insight into the unresolved evolutionary lineage of the bees, ants, and wasps?

At the same time, we investigated questions of functional genomics and epigenomics.
How many \textit{Polistes} genes show differential expression between the queen and worker castes?
What is the extent of alternative splicing in \textit{Polistes}, and do any alternative splicing events exhibit caste-related bias?
What is the extent of DNA methylation in \textit{Polistes}, and does the genome encode a full complement of methylation-related genes?

This last question led us to one of the highlight discoveries of the study: that \textit{Polistes} lacks a critical DNA methyltransferase (\textit{Dnmt3}) and has essentially no DNA methylation genome-wide.
But more in line with the focus of this dissertation, the \textit{Polistes dominula} genome project exposed me to the challenges of creating genomic data resources \textit{de novo} for a non-model research system, and the corresponding challenges of data quality assessment and management.
These challenges motivated the development of methods and tools to facilitate comparison, evaluation, and analysis of genome annotations, with additional applications to studying genome organization.

\section{ParsEval: comparison of distinct annotation sources}

Automated genome annotation typically relies on integrating tools for \textit{ab initio} gene prediction, transcript and protein spliced alignment, and evaluation of support for gene structural components.
Each class of tools comes with a variety of parameter settings, and it can be difficult to predict in advance the influence these parameters will have on the final annotation product.
%For example, we initially used gene prediction parameters trained for \textit{Drosophila melanogaster} during the preliminary stages of the \textit{Polistes dominula} annotation.
%After observing that these parameters produced results of inconsistent quality, we found that settings trained for \textit{Arabidopsis thaliana} performed more reliably, which in hindsight is not surprising considering that \textit{Arabidopsis} is much more similar to \textit{Polistes} than \textit{Drosophila} is in terms of nucleotide composition.
When annotating a non-model genome \textit{de novo}, it is often necessary to refine parameter selection on a small subset of the available data before proceeding to annotate the entire genome.
Subsequent re-annotation is often necessary when additional data (such as ESTs or RNA-Seq reads) become available, or when improved gene prediction methods are published.
And in some cases, the scientists annotating a particular genome have a vested interest in improving the accuracy of the annotation software itself.
In each of these scenarios, a primary objective is to identify similarities and differences between annotations derived from different parameter settings or alternative workflows, to facilitate evaluation of the annotations.

Development of a new annotation pipeline (CpGAT) within our research group provided the initial motivation for ParsEval.
Evaluating this tool's performance in comparison to a gold-standard annotation was impractical manually, and existing software tools \cite{Keibler,Wang} lacked important features such as locus-scale resolution.
I created the ParsEval program to address the need for genome-scale evaluations with locus-scale detail.
Later during the preliminary stages of the \textit{Polistes dominula} genome project, ParsEval proved a valuable data assessment tool as we tuned our genome annotation workflow.

The ParsEval paper introduces a precise operational definition for a \textit{gene locus} based on the locations of annotated gene models.
The objective was to define a parsing of the genome into distinct units that are complete and can be analyzed independently.
Generalizing this concept and applying it as an organizational principle for genome analyses has been a major focus of this dissertation.

\section{iLoci: an organizational framework}

The ParsEval tool proved valuable throughout the \textit{Polistes dominula} genome project, but we subsequently encountered issues that required additional attention and development.
Although our \textit{Polistes} assembly and annotation compared favorably to other published hymenopteran genomes, our data was subject to the same quality inconsistencies that characterize any NGS-based genome project.
Answering questions related to gene expression and genome composition therefore required careful consideration of, for example, how precisely to handle overlapping gene models and how to distinguish differences rooted in biology from technical artifacts.
It was in addressing these issues that we extended the \textit{gene locus} definition introduced by ParsEval and developed the \textit{interval locus (iLocus)} as a more generalized organizational framework for genome analyses.

iLoci define an unambiguous parsing of an annotated genome sequence into distinct regions, each encapsulating the genomic context of a gene or intergenic space.
This parsing provides a complete and granular decomposition of the genome, and characteristics of iLoci are reflective of a variety of genomic features of interest.
In parallel with the \textit{P. dominula} genome project in which iLoci played a prominent role, we investigated general applications of iLoci.
We discuss their utility for describing the gene content of a genome and for applying quality control when calculating diagnostic characteristics of a genome.
We demonstrate the utility of iLoci for characterizing genome organization, within a single genome and between multiple genomes in a clade of species.
Finally, we report the stability of iLoci across distinct assembly and annotation versions, highlighting their utility as reproducible units of analysis.

% \section{GeneAnnoLogy: annotation version control}
%
% Continual refinement of our genome assembly and annotation hindered progress on our \textit{P. dominula} research at times.
% This issue is certainly not unique to our work.
% For example, the honeybee \textit{Apis mellifera} (the model social insect) had three official annotation versions in concurrent use during our work on the \textit{Polistes} genome project, with some studies even making their own unpublished refinements to an annotation \cite{Dnmt3KD,TrueSight}.
% What seemed to be lacking was a precise way to refer to a particular gene or genomic region, as annotated at a particular time, and to make statements about its expression, or conservation, or methylation status, or any number of additional characteristics.
%
% The idea of tracking annotations over time is not new \cite{Eilbeck}, and some well-supported communities provide tools for mapping annotations from an older assembly version to updated assembly \cite{liftOver,PAGIT}.
% However, iLoci provide an alternative solution to these issues and, along with the additional benefits previously described, furnish a complete framework for organization, quality control, and reproducibility for provisional genome projects.
%
% At an early stage in my training, I prototyped a tool called GeneAnnoLogy for maintaining an annotation version history, leveraging existing version control tools to track changes to individual iLoci over time.
% I later built in features that facilitate filtering iLoci based on their length, nucleotide composition, annotation quality, or any number of additional characteristics that can be computed from or attached to the annotation.
% The GeneAnnoLogy tool encapsulates the culmination of concepts and principles investigated in this dissertation.

\section{Dissertation Organization}

This dissertation is organized into five chapters.
Chapter 1 provides an overview of the dissertation, a motivation for the work, and a brief discussion of relevant literature.
Chapters 2 through 4 are presented as self-contained manuscripts:
chapter 2 is a research paper published in \textit{Molecular Ecology} describing the genome, transcriptome, and methylome of the paper wasp \textit{Polistes dominula}, highlighting its reduced DNA methylation system, several hundred loci with caste-related differential expression, and the lack of any detectable caste-related differential splicing in the adult organism;
chapter 3 is a paper published in \textit{BMC Bioinformatics} describing \textit{ParsEval}, a tool for comparing two alternate sources of annotation for a genome sequence;
chapter 4 is a methodology paper to be submitted to \textit{Genome Biology}, describing the use of \textit{interval loci (iLoci)} as an organizational framework for reproducible genome analysis;
%chapter 5 is a manuscript slated for submission to \textit{A Journal} describing \textit{GeneAnnoLogy}, an \textit{iLocus}-based tool for quality control and version control of genome annotations.
Chapter 5 provides brief concluding remarks and suggestions for further research.
